src/main/java/graphql/schema/idl/SchemaTransformer.java

```diff
@@ -0,0 +1,19 @@
+package graphql.schema.idl;
+
+import graphql.schema.GraphQLSchema;
+
+/**
+ * These are called by the {@link SchemaGenerator} after a valid schema has been built
+ * and they can then adjust it accordingly with some sort of post processing.
+ */
+public interface SchemaTransformer {
+
+    /**
+     * Called to transform the schema from its built state into something else
+     *
+     * @param originalSchema the original built schema
+     *
+     * @return a non null schema
+     */
+    GraphQLSchema transform(GraphQLSchema originalSchema);
+}
```

Comment:
As discussed perhaps we should put a `transform` method on `GraphQLSchema` that takes a visitor and we can do the setup

Reply:
we already have a `transform` inside GraphQLSchema which only transforms `GraphQLSchema` and is therefore limited. Add another `transform` or give it another name?